x-pack/plugins/watcher/server/models/watch/threshold_watch/build_visualize_query.js

```diff
@@ -117,11 +106,15 @@ export function buildVisualizeQuery(watch, visualizeOptions) {
     termOrder,
   });
   const body = watchInput.search.request.body;
-  const dateAgg = buildDateAgg({
-    field: watch.timeField,
-    interval: visualizeOptions.interval,
-    timeZone: visualizeOptions.timezone,
-  });
+  const dateAgg = {
+    date_histogram: {
+      field: watch.timeField,
+      interval: visualizeOptions.interval, // Only used in 7.x, it will be undefined in 8.x
```

Comment:
I think it will be clearer if the code is explicit about what's defined and what isn't. Here's what I'm thinking, incorporating my suggestion from my previous comment about doing the differentiation here at the last possible opportunity (using pseudocode):

```js
  if (7) {
    dateAgg.date_histogram.interval = visualizeOptions.interval;
  }

  if (8) {
    dateAgg.date_histogram.fixed_interval = visualizeOptions.interval;
  }
```

WDYT?

Reply:
I prefer to only have if condition branches for the deprecated code, the default behaviour behind what's in master. This way it will be a clear cut and we'll only delete code whenever we want to clean branches we don't support anymore (7.x once in 9.x)

As you can see here: https://github.com/elastic/kibana/pull/113527/commits/3eedaadf2982bc894d2b4e814efa0be3527c1d25#diff-416a826acd66e06ceeb33c65dae647f5e5343b71dc5b98353a32c144bd0c3d48R122